python/cudf/cudf/core/column/datetime.py

```diff
@@ -78,19 +77,8 @@ def from_numpy(cls, array):
 
         mask = None
         if np.any(np.isnat(array)):
-            null = cudf.core.column.column_empty_like(
-                array, masked=True, newsize=1
-            )
-            col = libcudfxx.replace.replace(
-                as_column(Buffer(array), dtype=array.dtype),
-                as_column(
-                    Buffer(
-                        np.array([np.datetime64("NaT")], dtype=array.dtype)
-                    ),
-                    dtype=array.dtype,
-                ),
-                null,
-            )
+            col = as_column(Buffer(array), dtype=array.dtype)
+            col = utils.time_col_replace_nulls(col)
```

Comment:
Can we just remove this entire `from_numpy` function and move all of its logic into the `as_column` factory?